### PR TITLE
Hook tests

### PR DESCRIPTION
**Pull request by tk-vincit**
- - -
This change is [Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-ui/337)

### CODE REVIEW

**Review by tk-vincit — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-ui/337)* status: 0 of 25 files reviewed, 1 unresolved discussion

___
*[`.github/workflows/run-integration-tests.yml` line 22 at r4](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NCyf5ATBNxuwk7HkmIO:-NCyf5ATBNxuwk7HkmIP:b-4r9pxe) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fc60950f7297e3e39fd54b6932fadd10b5a5e7fc/.github/workflows/run-integration-tests.yml#L22)):*
> ```YAML
> 
>       - name: start e2e env
>         uses: HSLdevcom/jore4-flux/github-actions/setup-e2e-environment@setup-e2e-environment-v1
> ```

Wondering if we could optimize this easily by spinning up only required subset ot dependencies?

**Review by pallost — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-ui/337)* status: 0 of 26 files reviewed, 13 unresolved discussions (waiting on @pallost and @tk-vincit)

___
*[`Dockerfile` line 11 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCtqa_Eb6HiiemP83H:-NDCtqa_Eb6HiiemP83I:b-5qxmo4) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/Dockerfile#L11)):*
> ```Dockerfile
> COPY ./ui/public ./ui/public
> COPY ./ui/tsconfig.json ./ui/next.config.js ./ui/next-env.d.ts ./ui/tailwind.config.js ./ui/postcss.config.js ./ui/convert-theme-to-ts.js ./ui/theme.js ./ui/graphql.schema.json ./ui/
> COPY ./test-db-manager/rollup.config.js ./test-db-manager/tsconfig.json ./test-db-manager/
> ```

wow, this is bad... well, let's try to get rid of this soon, but no need to do this in this PR
___
*[`package.json` line 23 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCu4h-270UtU5WUrSs:-NDCu4h-270UtU5WUrSt:b-sngxd0) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/package.json#L23)):*
> ```JSON
>     "test:e2e": "yarn ws:db build && yarn ws:e2e cypress run --browser chrome",
>     "test:e2e:open": "yarn ws:db build && yarn ws:e2e cypress open",
>     "qa": "yarn --frozen-lockfile && yarn ws:db build && yarn ts:check && yarn prettier:check && yarn lint && yarn test && yarn test:e2e"
> ```

let's add yarn test:integration to the qa script as well. It's anyway running cypress tests already so could just run integration tests as well...
___
*[`.github/workflows/run-integration-tests.yml` line 22 at r4](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NCyf5ATBNxuwk7HkmIO:-NDCuQ5KBuXISvt73dLO:b604h93) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fc60950f7297e3e39fd54b6932fadd10b5a5e7fc/.github/workflows/run-integration-tests.yml#L22)):*
<details><summary><i>Previously, tk-vincit wrote…</i></summary><blockquote>

Wondering if we could optimize this easily by spinning up only required subset ot dependencies?
</blockquote></details>

it wouldn't be signifantly faster: the slowest microservice to start is hasura itself, which you anyway need in this test. But I've been thinking of adding a parameter to setup-e2e-environment github action to be able to include/exclude services
___
*[`.github/workflows/run-integration-tests.yml` line 25 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCuq7981d0t58_Dj2D:-NDCuq7981d0t58_Dj2E:bfmdm7y) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/.github/workflows/run-integration-tests.yml#L25)):*
> ```YAML
> 
>       - name: Verify that all services are up and running
>         uses: HSLdevcom/jore4-tools/github-actions/healthcheck@healthcheck-v1
> ```

the setup-e2e-environment action is already checking that all the services are up and running, no need to do it separately here
___
*[`ui/src/graphql/client.ts` line 18 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCxH_T5zikMaGXDmTF:-NDCxH_T5zikMaGXDmTG:blo6lmy) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/graphql/client.ts#L18)):*
> ```XML
> import { authRoleMiddleware } from './auth';
> 
> const isTesting = process.env.NODE_ENV === 'test';
> ```

you should write a comment here that jest automatically sets this environment variable when running its tests, there's no need to set it manually
___
*[`ui/src/graphql/client.ts` line 55 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCydyo3ZaBrOusaeRM:-NDCydyo3ZaBrOusaeRN:b-ooc15b) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/graphql/client.ts#L55)):*
> ```XML
> };
> 
> const getHttpLinkOptions = (uri: string) => {
> ```

this is getting messy here, really hard to follow that what links get created in case what flags are present. Also, I don't think that we need a new flag: the isBrowser and the !isTesting pretty much mean the same of us. An idea to clarify this:

```
const buildWebSocketLink = (graphqlUrlPath: string) => {
  return new WebSocketLink({
        // WebSocketLink doesn't work with relative url's, so we have to
        // turn relative url into absolute.
        uri: `${mapHttpToWs(window.location.origin)}${graphqlUrlPath}`,
        options: {
          reconnect: true,
        },
      })
}

const buildHttpLink = (graphqlUrlPath: string, isTesting: boolean) => {
  const httpLinkConfig = isTesting ? {
     fetch: require('cross-fetch'),
     // logging in directly as admin, only works in e2e environment
     headers: { 'x-hasura-admin-secret': 'hasura' },
     uri: `http://localhost:3300${graphqlUrlPath}`,
  } : {
     uri: graphqlUrlPath,
  }
}

// I don't think there's a difference for us in isBrowser and !isTesting, you probably could just join these two flags
const buildConnectionLink = (isBrowser: boolean, isTesting: boolean) => {
  const graphqlUrlPath = '/api/graphql/v1/graphql';

  const httpLink = buildHttpLink(graphqlUrlPath, isTesting);
  if (isTesting || !isBrowser) {
    return httpLink;
  }

  const wsLink = buildWebSocketLink(graphqlUrlPath);
  return split(..., wsLink, httpLink);
}
```
___
*[`ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts` line 31 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD1aOp8hW4yYAO0aew:-NDD1aOp8hW4yYAO0aex:b-ggdk3m) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts#L31)):*
> ```XML
>     validity_start: DateTime.fromISO('2022-08-11T13:08:43.315+03:00'),
>     validity_end: DateTime.fromISO('2023-08-11T13:08:43.315+03:00'),
>   },
> ```

instead of using spread operators, I was dreaming of something like this:

```
const lines = [
  buildLine({label: 'indefinite', line_id: '...'}),
  buildLine({label: '65x', line_id: '...', validity_start: '...', validity_end: '...'),
]

This pattern would be IMHO easier to follow when expanding on it later:

const lines = [
  buildLineWithRoute({label: '65', route: {label: '65x'}}),
  buildIndefiniteLine({label: 'indefinite'}),
  buildTramLine({label: 'tram'}),
  buildLine({label: '65', line_routes: {data: [buildRoute({label: '65x'})]}})
]
```

Also, should we make line_id also a required parameter besides the label?

... but this is not part of the PR
___
*[`ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts` line 43 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD3MnK4IKmgFDWjyAW:-NDD3MnL5uiufat8HjpT:b-pprtw8) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts#L43)):*
> ```XML
>   validityEnd,
>   priority = Priority.Standard,
> }: {
> ```

you already had a `CommonParams` type in `useCheckValidityAndPriorityConflicts.ts`
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 30 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD4gPZDpb7UKM-VqLH:-NDD4gPZDpb7UKM-VqLI:b-jg5dqx) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L30)):*
> ```XML
> 
> // eslint-disable-next-line @typescript-eslint/no-explicit-any
> const responseLogger = (message: string, res: any) => {
> ```

didn't ExplicitAny work here?
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 33 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD4kT65381d1I0jq2i:-NDD4kT65381d1I0jq2j:begl1a) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L33)):*
> ```XML
>   if (res.errors) {
>     // eslint-disable-next-line no-console
>     console.log(`${message}:`, res);
> ```

This is really misunderstandable, you only log the response if there's an error. Please rename this function to e.g. `logOnError` or something similar
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 164 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD6Isu5UfvDm4pkK9S:-NDD6Isu5UfvDm4pkK9T:btw8oq4) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L164)):*
> ```XML
>     const mutation = mapToDeleteInfraLinksAlongRouteMutation(
>       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
>       infraLinksAlongRoute.map((item) => item.infrastructure_link_id!),
> ```

let's fix the type and add `& {xxx_id: UUID}` to make sure the entity has an id!
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 173 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD6QGk54u18tL4xX3s:-NDD6QGl2jvjcgd7SroR:btw8oq4) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L173)):*
> ```XML
>     const mutation = mapToDeleteJourneyPatternsMutation(
>       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
>       journeyPatterns.map((item) => item.journey_pattern_id!),
> ```

let's fix the type and add `& {xxx_id: UUID}` to make sure the entity has an id!
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 182 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD6Qy22nXOtF6zO64O:-NDD6Qy22nXOtF6zO64P:btw8oq4) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L182)):*
> ```XML
>     const mutation = mapToDeleteStopsInJourneyPatternMutation(
>       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
>       stopsInJourneyPattern.map((item) => item.journey_pattern_id!),
> ```

let's fix the type and add `& {xxx_id: UUID}` to make sure the entity has an id!

**Review by tk-vincit — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-ui/337)* status: 0 of 30 files reviewed, 13 unresolved discussions (waiting on @pallost and @tk-vincit)

___
*[`package.json` line 23 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCu4h-270UtU5WUrSs:-NDRQ7He1SMQXTWNOOVT:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/package.json#L23)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

let's add yarn test:integration to the qa script as well. It's anyway running cypress tests already so could just run integration tests as well...
</blockquote></details>

Done.
___
*[`.github/workflows/run-integration-tests.yml` line 25 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCuq7981d0t58_Dj2D:-NDRQo3-0gNWUDTGTZhF:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/.github/workflows/run-integration-tests.yml#L25)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

the setup-e2e-environment action is already checking that all the services are up and running, no need to do it separately here
</blockquote></details>

Done.
___
*[`ui/src/graphql/client.ts` line 18 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCxH_T5zikMaGXDmTF:-NDRQy2M-KMZjtjvarFb:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/graphql/client.ts#L18)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

you should write a comment here that jest automatically sets this environment variable when running its tests, there's no need to set it manually
</blockquote></details>

Done.
___
*[`ui/src/graphql/client.ts` line 55 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDCydyo3ZaBrOusaeRM:-NDRl8oG6evLOIRKJ8kl:b-46d92r) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/graphql/client.ts#L55)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

this is getting messy here, really hard to follow that what links get created in case what flags are present. Also, I don't think that we need a new flag: the isBrowser and the !isTesting pretty much mean the same of us. An idea to clarify this:

```
const buildWebSocketLink = (graphqlUrlPath: string) => {
  return new WebSocketLink({
        // WebSocketLink doesn't work with relative url's, so we have to
        // turn relative url into absolute.
        uri: `${mapHttpToWs(window.location.origin)}${graphqlUrlPath}`,
        options: {
          reconnect: true,
        },
      })
}

const buildHttpLink = (graphqlUrlPath: string, isTesting: boolean) => {
  const httpLinkConfig = isTesting ? {
     fetch: require('cross-fetch'),
     // logging in directly as admin, only works in e2e environment
     headers: { 'x-hasura-admin-secret': 'hasura' },
     uri: `http://localhost:3300${graphqlUrlPath}`,
  } : {
     uri: graphqlUrlPath,
  }
}

// I don't think there's a difference for us in isBrowser and !isTesting, you probably could just join these two flags
const buildConnectionLink = (isBrowser: boolean, isTesting: boolean) => {
  const graphqlUrlPath = '/api/graphql/v1/graphql';

  const httpLink = buildHttpLink(graphqlUrlPath, isTesting);
  if (isTesting || !isBrowser) {
    return httpLink;
  }

  const wsLink = buildWebSocketLink(graphqlUrlPath);
  return split(..., wsLink, httpLink);
}
```
</blockquote></details>

Fixed this.

Oh, I still have to check if we need isTesting flag.
___
*[`ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts` line 43 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD3MnK4IKmgFDWjyAW:-NDRWII309SuiPDXi6rU:buc3alj) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts#L43)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

you already had a `CommonParams` type in `useCheckValidityAndPriorityConflicts.ts`
</blockquote></details>

Good point. Done.

...or actually, it doesn't work so well here as we want to have 'priority' as optional to keep tests cleaner. But how about now?
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 30 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD4gPZDpb7UKM-VqLH:-NDRWdl87M-wBOvPZXUS:b6ctgnt) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L30)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

didn't ExplicitAny work here?
</blockquote></details>

It does, but write it like this as now this indicates that there might be better type available for this... But OK, as we are not doing anything else than logging the response, maybe I'll switch this to ExplicitAny. Done.
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 33 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD4kT65381d1I0jq2i:-NDRX-Hf6BgVuYL7l1VP:b-oadpkn) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L33)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

This is really misunderstandable, you only log the response if there's an error. Please rename this function to e.g. `logOnError` or something similar
</blockquote></details>

As you wish. Done.
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 164 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD6Isu5UfvDm4pkK9S:-NDRZqg6Abvbld1JwdRS:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L164)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

let's fix the type and add `& {xxx_id: UUID}` to make sure the entity has an id!
</blockquote></details>

Done.
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 173 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD6QGk54u18tL4xX3s:-NDRhcgCCANRqAkuap-T:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L173)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

let's fix the type and add `& {xxx_id: UUID}` to make sure the entity has an id!
</blockquote></details>

Done.
___
*[`ui/src/utils/test-utils/integration-tests/db.ts` line 182 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD6Qy22nXOtF6zO64O:-NDRhdLPC54AqPO8drMV:b-896fix) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/utils/test-utils/integration-tests/db.ts#L182)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

let's fix the type and add `& {xxx_id: UUID}` to make sure the entity has an id!
</blockquote></details>

Done.

**Review by pallost — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-ui/337)* status: 0 of 30 files reviewed, 4 unresolved discussions (waiting on @pallost and @tk-vincit)

___
*[`ui/src/graphql/client.ts` line 52 at r11](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDSL3i21NcEe10gPylZ:-NDSL3i21NcEe10gPyl_:b-55mt6t) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/501e0176bd48a5337fd617907a4337447210e12c/ui/src/graphql/client.ts#L52)):*
> ```XML
>     // WebSocketLink doesn't work with relative url's, so we have to
>     // turn relative url into absolute.
>     uri: `${mapHttpToWs(window.location.origin)}${graphqlUrlPath}`,
> ```

it you want, you could actually readd that `getGraphqlServerUrl(isBrowser, isTesting)` function to simplify this
___
*[`ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts` line 43 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD3MnK4IKmgFDWjyAW:-NDSLMCc71hIuT5kDXZb:bfu6ox4) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts#L43)):*
<details><summary><i>Previously, tk-vincit wrote…</i></summary><blockquote>

Good point. Done.

...or actually, it doesn't work so well here as we want to have 'priority' as optional to keep tests cleaner. But how about now?
</blockquote></details>

not much prettier :D I wish we had an `OptionalKeys<>` type that works similar to `RequiredKeys<>`
your call: leave it inline or use `OptionalKeys`

**Review by tk-vincit — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/hsldevcom/jore4-ui/337)* status: 0 of 30 files reviewed, 4 unresolved discussions (waiting on @pallost and @tk-vincit)

___
*[`ui/src/graphql/client.ts` line 52 at r11](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDSL3i21NcEe10gPylZ:-NDSNVktAOzwJMjNuNwv:b7k0sqj) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/501e0176bd48a5337fd617907a4337447210e12c/ui/src/graphql/client.ts#L52)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

it you want, you could actually readd that `getGraphqlServerUrl(isBrowser, isTesting)` function to simplify this
</blockquote></details>

I can re-add it, but not sure if it improves readability in this specific case. Using absolute url instead of relative is needed in this case specificly because of limitations of `WebSockerLink`, and now that info is at least coupled with the usage of WebSockerLink. 
___
*[`ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts` line 43 at r8](https://reviewable.io/reviews/hsldevcom/jore4-ui/337#-NDD3MnK4IKmgFDWjyAW:-NDSOlb_-mv23FugcfjP:bx9wok3) ([raw file](https://github.com/hsldevcom/jore4-ui/blob/fb8a6c7a1931620037f2d402659a7d57f54318a5/ui/src/hooks/useCheckValidityAndPriorityConflicts.integration.spec.ts#L43)):*
<details><summary><i>Previously, pallost (Tamas Pallos) wrote…</i></summary><blockquote>

not much prettier :D I wish we had an `OptionalKeys<>` type that works similar to `RequiredKeys<>`
your call: leave it inline or use `OptionalKeys`
</blockquote></details>

Oh, didn't remember that we had that. Fixed now :)
